include/nbl/video/alloc/SubAllocatedDescriptorSet.h

```diff
@@ -0,0 +1,366 @@
+			m_addresses = other.m_addresses;
+			m_binding = other.m_binding;
+			return *this;
+		}
+
+		// This is needed for the destructor of TimelineEventHandlerST
+		// Don't call this directly
+		// TODO: Find a workaround for this
+		inline void operator()()
+		{
+			assert(false); // This should not be called, timeline needs to be drained before destructor
+			// core::vector<IGPUDescriptorSet::SDropDescriptorSet> nulls(m_addresses->size());
+			// auto ptr = nulls.data();
+			// operator()(ptr);
+			// auto size = ptr - nulls.data();
+			// m_composed->m_logicalDevice->nullifyDescriptors({nulls.data(),size_type(size)});
+		}
+
+		// Takes count of allocations we want to free up as reference, true is returned if
+		// the amount of allocations freed was >= allocationsToFreeUp
+		// False is returned if there are more allocations to free up
+		inline bool operator()(size_type& allocationsToFreeUp, IGPUDescriptorSet::SDropDescriptorSet* &outNullify)
+		{
+			auto prevNullify = outNullify;
+			operator()(outNullify);
+			auto totalFreed = outNullify-prevNullify;
+
+			// This does the same logic as bool operator()(size_type&) on 
+			// CAsyncSingleBufferSubAllocator
+			bool freedEverything = totalFreed >= allocationsToFreeUp;
+		
+			if (freedEverything) allocationsToFreeUp = 0u;
+			else allocationsToFreeUp -= totalFreed;
+			return freedEverything;
+		}
+	protected:
+		core::smart_refctd_dynamic_array<value_type> m_addresses;
+		SubAllocatedDescriptorSet* m_composed; // TODO: shouldn't be called `composed`, maybe `parent` or something
+		uint32_t m_binding;
+	};
+	using EventHandler = MultiTimelineEventHandlerST<DeferredFreeFunctor>;
+protected:
+	struct SubAllocDescriptorSetRange {
+		EventHandler eventHandler = EventHandler({});
+		std::unique_ptr<AddressAllocator> addressAllocator = nullptr;
+		std::unique_ptr<ReservedAllocator> reservedAllocator = nullptr;
+		size_t reservedSize = 0;
+		asset::IDescriptor::E_TYPE descriptorType = asset::IDescriptor::E_TYPE::ET_COUNT;
+
+		SubAllocDescriptorSetRange(
+			std::unique_ptr<AddressAllocator>&& inAddressAllocator,
+			std::unique_ptr<ReservedAllocator>&& inReservedAllocator,
+			size_t inReservedSize,
+			asset::IDescriptor::E_TYPE inDescriptorType) :
+			eventHandler({}), addressAllocator(std::move(inAddressAllocator)),
+			reservedAllocator(std::move(inReservedAllocator)), 
+			reservedSize(inReservedSize),
+			descriptorType(inDescriptorType) {}
+		SubAllocDescriptorSetRange() {}
+
+		SubAllocDescriptorSetRange& operator=(SubAllocDescriptorSetRange&& other)
+		{
+			addressAllocator = std::move(other.addressAllocator);
+			reservedAllocator = std::move(other.reservedAllocator);
+			reservedSize = other.reservedSize;
+			descriptorType = other.descriptorType;
+
+			// Nullify other
+			other.addressAllocator = nullptr;
+			other.reservedAllocator = nullptr;
+			other.reservedSize = 0u;
+			other.descriptorType = asset::IDescriptor::E_TYPE::ET_COUNT;
+			return *this;
+		}
```

Comment:
as a matter of good practice you should `=delete` the `const&` assignment and constructor (and really any ctor/assignment you're not using)